server/routes/pt.js

```diff
@@ -114,4 +116,184 @@ router.post('/search', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+    const doc = await Pt.findById(id);
+    if (!doc) return res.status(NOT_FOUND).send();
+
+    const requests = doc.requests;
+    const names = [];
+    for await (const docs of User.findById(requests)) {
+      try {
+        const user = await User.findById(docs._id);
+        names.push(user.publicName);
+      }
+      catch (err) {
+        console.error('/getrequests found missing User doc: ' +
+        err.message)
+      }
+    }
+
+    return res.status(OK).send(names);
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/acceptrequest', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+
+    const doc = await Pt.findById(id);
+    const requests = doc.requests;
+    
+    for await (const doc of User.findById(requests)) {
+      try {
+        if(doc.publicName == req.body.name) {
+        
+          // Update the PT's clients array to add the requesting client
+          Pt.findByIdAndUpdate(id,
+            {$push: {clients: doc._id}},
+            {safe: true, upsert: true},
+            function(err,doc) {
+              if(err) {
+                console.log(err);
+              }
+            });
+
+          // Delete the client from the PT's request array
+          Pt.findByIdAndUpdate(id,
+            {$pull: {requests: doc._id}},
+            {safe: true, upsert: true},
+            function(err,doc) {
+              if(err) {
+                console.log(err);
+              }
+            });
+
+          // Update the client's PT
+          const clientDoc = await Client.findByIdAndUpdate(doc._id);
+          clientDoc.pt = id;
+          clientDoc.save();
+
+          // Delete the PT from the client's requests array
+          Client.findByIdAndUpdate(doc._id,
+            {$pull: {requests: id}},
+            {safe: true, upsert: true},
+            function(err,doc) {
+              if(err) {
+                console.log(err);
+              }
+            });
+
+          return res.status(OK);
+        }
+      }
+      catch (err) {
+        return res.status(BAD_REQUEST).send(err.message);
+      }
+  
+    }
+  return res.status(NOT_FOUND).send(err.message);
+  }
+  catch(err) {
+    return res.status(NOT_FOUND).send(err.message); 
+  }
+   
+});
+
+router.post('/deleterequest', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+
+    const doc = await Pt.findById(id);
+    const requests = doc.requests;
+    
+    for await (const doc of User.findById(requests)) {
+      try {
+        if(doc.publicName == req.body.name) {
+      
+          // Delete the client from the PT's request array
+          Pt.findByIdAndUpdate(id,
+            {$pull: {requests: doc._id}},
+            {safe: true, upsert: true},
+            function(err,doc) {
+              if(err) {
+                console.log(err);
```

Comment:
```suggestion
                console.error(err);
```
For these, logging the error and not sending a failure response makes sense.